osbs/build/spec.py

```diff
@@ -397,37 +435,116 @@ def set_params(self, git_uri=None, git_ref=None,
-        # returns the same values throughout the life of the builder.
-        # Before each `randrange` call we should be calling `.seed` to prevent this
-        random.seed()
+    def __repr__(self):
+        return "Spec(%s)" % self.__dict__
 
-        tag_segments = [
-            self.koji_target.value or 'none',
-            str(random.randrange(10**(RAND_DIGITS - 1), 10**RAND_DIGITS)),
-            timestamp
+
+class BuildUserParams(BuildCommon):
+    def __init__(self):
+        # defines image_tag, koji_target, platform, arrangement_version
+        super(BuildUserParams, self).__init__()
+        self.git_uri = BuildParam('git_uri')
+        self.git_ref = BuildParam('git_ref', default=DEFAULT_GIT_REF)
+        self.git_branch = BuildParam('git_branch')
+        self.user = UserParam()
+        self.component = BuildParam('component')
+        self.name = BuildIDParam()
+        self.build_image = BuildParam('build_image')
+        self.build_imagestream = BuildParam('build_imagestream')
+        self.build_from = BuildParam('build_from')
+        self.trigger_imagestreamtag = BuildParam('trigger_imagestreamtag')
+        self.imagestream_name = BuildParam('imagestream_name')
+        self.platforms = BuildParam("platforms", allow_none=True)
+        self.release = BuildParam("release", allow_none=True)
+        self.koji_task_id = BuildParam("koji_task_id", allow_none=True)
+        self.flatpak = BuildParam("flatpak", default=False)
+        self.flatpak_base_image = BuildParam("flatpak_base_image", allow_none=True)
+        self.signing_intent = BuildParam("signing_intent", allow_none=True)
+        self.yum_repourls = BuildParam("yum_repourls")
+        self.compose_ids = BuildParam("compose_ids", allow_none=True)
+        self.reactor_config_map = BuildParam("reactor_config_map", allow_none=True)
+        # BuildUserParams is only used with arrangement_version 6
+        self.arrangement_version.value = 6
+        self.required_params = [
+            self.git_uri,
+            self.git_ref,
+            self.user,
+            self.koji_target,
         ]
 
-        # Support for platform specific tags has only been added in arrangement 4.
-        if self.platform.value and (self.arrangement_version.value or 0) >= 4:
-            tag_segments.append(self.platform.value)
+    def set_params(self,
+                   git_uri=None, git_ref=None, git_branch=None,
+                   base_image=None, name_label=None,
+                   user=None,
+                   component=None, release=None,
+                   build_image=None, build_imagestream=None, build_from=None,
+                   platforms=None, platform=None,
+                   koji_target=None, koji_task_id=None,
+                   flatpak=None, flatpak_base_image=None,
+                   reactor_config_map=None,
+                   yum_repourls=None, signing_intent=None, compose_ids=None,
```

Comment:
Needs to accept `filesystem_koji_task_id` as well and store this is `self.filesystem_koji_task_id.value`, and of course `__init__` needs to create `self.filesystem_koji_task_id = BuildParam(...)` like `BuildSpec.__init__` does. Then `BuildRequestV2` needs to pass that value through in its call to this method.

Also do we have anything testing that the value ends up correctly in the add_filesystem plugin?

Reply:
Fixed.